contracts/IndexStaking.sol

```diff
@@ -0,0 +1,56 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.6.6;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../interfaces/IIndexHybridToken.sol";
+import "../interfaces/IHybridToken.sol";
+import "../libraries/SafeTransfer.sol";
+
+contract IndexStaking {
+    using SafeMath for uint256;
+
+    address public stakingToken;
+    address public rewardToken;
+
+    // init T, S, stake, S0
+    uint public staked; // S
+    uint public activeStakeDeposits; // T
+    mapping(address => uint) public stake; // stake
+    mapping(address => uint) public stakedSnapshot; // S0
+
+    constructor(address _stakingTokenAddress, address _rewardTokenAddress) public {
+        stakingToken = _stakingTokenAddress;
+        rewardToken = _rewardTokenAddress;
+    }
+
+    function deposit(uint _amount) external {
+        _update();
```

Comment:
You should remove `_update()` form this method. It is essential just for withdrawal moment.